lib/matplotlib/pyplot.py

```diff
@@ -542,6 +530,11 @@ def make_active(event):
 
         _pylab_helpers.Gcf.set_active(figManager)
         figManager.canvas.figure.number = num
+        if _INSTALL_FIG_OBSERVER:
+            def auto_draw(fig):
+                if fig.stale and matplotlib.is_interactive():
+                    fig.canvas.draw_idle()
+            figManager.canvas.figure.add_callback(auto_draw)
```

Comment:
As this is registered on a per-figure basis that gets a bit messy book keeping wise and would probably need to be pushed back up into Gcf which is more complicated than I _really_ want to make this.  I am happy to say that if you _really_ care about controlling ion/ioff, use IPython where we have better control of the hooks.

In the base case, all new figures will respect the setting, existing figures will respect the last setting which seems fair-ish.

I also don't see the use case for rapidly switching between ion/ioff.